examples/AdvancedOperations/AddDynamicSearchAds.php

```diff
@@ -0,0 +1,351 @@
+    }
+
+    /**
+     * Creates a campaign budget.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the client customer ID
+     * @return string the campaign budget resource name
+     */
+    private static function createCampaignBudget(GoogleAdsClient $googleAdsClient, $customerId)
+    {
+        $campaignBudget = new CampaignBudget([
+            'name' => new StringValue(['value' => 'Interplanetary Cruise Budget #' . uniqid()]),
+            'delivery_method' => BudgetDeliveryMethod::STANDARD,
+            'amount_micros' => new Int64Value(['value' => 500000])
+        ]);
+
+        // Creates a campaign budget operation.
+        $campaignBudgetOperation = new CampaignBudgetOperation();
+        $campaignBudgetOperation->setCreate($campaignBudget);
+     
+        // Issues a mutate request to add campaign budgets.
+        $campaignBudgetServiceClient = $googleAdsClient->getCampaignBudgetServiceClient();
+        /** @var MutateCampaignBudgetsResponse $campaignBudgetResponse */
+        $campaignBudgetResponse = $campaignBudgetServiceClient->mutateCampaignBudgets(
+            $customerId,
+            [$campaignBudgetOperation]
+        );
+
+        $campaignBudgetResourceName = $campaignBudgetResponse->getResults()[0]->getResourceName();
+
+        printf("Added budget named '%s'.%s", $campaignBudgetResourceName, PHP_EOL);
+
+        return $campaignBudgetResourceName;
+    }
+
+    /**
+     * Creates a campaign.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the client customer ID without hyphens
+     * @param string $campaignBudgetResourceName the resource name of the campaign budget
+     * @return string the resource name of the newly created campaign
+     */
+    private static function createCampaign(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        string $campaignBudgetResourceName
+    ) {
+        $startDate = new StringValue(['value' => date('Ymd', strtotime('+1 day'))]);
+        $endDate = new StringValue(['value' => date('Ymd', strtotime('+1 month'))]);
+
+        $campaign = new Campaign([
+            'name' => new StringValue(['value' => 'Interplanetary Cruise #' . uniqid()]),
+            'advertising_channel_type' => AdvertisingChannelType::SEARCH,
+            'status' => CampaignStatus::PAUSED,
+            'manual_cpc' => new ManualCpc(),
+            'campaign_budget' => $campaignBudgetResourceName,
+            // Enables the campaign for DSAs.
+            'dynamic_search_ads_setting' => new DynamicSearchAdsSetting([
+                'domain_name' => new StringValue(['value' => 'example.com']),
+                'language_code' => new StringValue(['value' => 'en'])
+            ]),
+            // Optional: Sets the start and end dates.
+            'start_date' => $startDate,
```

Comment:
What about putting these date values inline instead of using local variables?

Reply:
Done.